azure-pipelines-templates/cfs/standard-service-ci-build.yml

```diff
@@ -0,0 +1,152 @@
+parameters:
+- name: DotNetCoreSdkVersion
+  type: string
+  default: '3.1.101'
+- name: ApiFrameworkVersion
+  type: string
+  default: 'netcoreapp3.1'
+- name: FunctionFrameworkVersion
+  type: string
+  default: 'netcoreapp3.1'
+- name: ProjectName
+  type: string
+  default: ''
+- name: BuildConfiguration
+  type: string
+  default: 'Release'
+- name: vstsFeed
+  type: string
+  default: 'ecf37bc0-472b-47bf-91e3-6774809fd194'
+- name: BuildApi
+  type: boolean
+  default: true
+- name: BuildFunction
+  type: boolean
+  default: true
+
+steps:
+ - task: UseDotNet@2
+   displayName: 'Use .NET Core sdk ${{ parameters.DotNetCoreSdkVersion }}'
+   inputs:
+     version: ${{ parameters.DotNetCoreSdkVersion }}
+
+ - task: DotNetCoreCLI@2
+   displayName: 'dotnet restore'
+   inputs:
+     command: restore
+     vstsFeed: '${{ parameters.vstsFeed }}'
+     projects: |
+      **/*.${{ parameters.ProjectName }}.*Tests/*.csproj
+      **/*.Models.*UnitTests/*.csproj
+      CalculateFunding.Api.${{ parameters.ProjectName }}/CalculateFunding.Api.{{ parameters.ProjectName }}.csproj
+      CalculateFunding.Functions.${{ parameters.ProjectName }}/CalculateFunding.Functions.{{ parameters.ProjectName }}.csproj
+
+- task: DotNetCoreCLI@2
+  displayName: 'dotnet test for All'
+  inputs:
+    command: test
+    projects: '**/*.Models.*UnitTests/*.csproj'
+    arguments: '--configuration $(releaseconfiguration) /p:AssemblyVersion=$(Build.BuildNumber)'
+
+ - task: DotNetCoreCLI@2
+   displayName: 'dotnet test for Service and Functions'
```

Comment:
I'd be tempted to merge the tasks for "dotnet test for All" and "dotnet test for Service and Functions" as there's no condition and it'll speed up the build slightly